packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1692,35 +1699,6 @@ export class Domain extends DomainBase implements IDomain {
         domainName: this.domainEndpoint,
       });
     }
-
-    const accessPolicyStatements: iam.PolicyStatement[] | undefined = unsignedBasicAuthEnabled
-      ? (props.accessPolicies ?? []).concat(unsignedAccessPolicy)
-      : props.accessPolicies;
-
-    if (accessPolicyStatements != null) {
-      const accessPolicy = new ElasticsearchAccessPolicy(this, 'ESAccessPolicy', {
```

Comment:
**If** you are able to not use `ElasticsearchAccessPolicy` then I think it would be good to delete that whole custom resource and the corresponding tests.